surefire-its/src/test/java/org/apache/maven/surefire/its/jiras/Surefire2117XmlReportingNestedIT.java

```diff
@@ -0,0 +1,120 @@
+package org.apache.maven.surefire.its.jiras;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.maven.surefire.its.fixture.OutputValidator;
+import org.apache.maven.surefire.its.fixture.SurefireJUnit4IntegrationTestCase;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Integration Test for SUREFIRE-2117
+ */
+@RunWith( Parameterized.class )
+@SuppressWarnings( "checkstyle:magicnumber" )
+public class Surefire2117XmlReportingNestedIT extends SurefireJUnit4IntegrationTestCase
+{
+    @Parameter
+    @SuppressWarnings( "checkstyle:visibilitymodifier" )
+    public String platform;
+
+    @Parameter( 1 )
+    @SuppressWarnings( "checkstyle:visibilitymodifier" )
+    public String jupiter;
+
+    @Parameter( 2 )
+    @SuppressWarnings( "checkstyle:visibilitymodifier" )
+    public String opentest;
+
+    @Parameter( 3 )
+    @SuppressWarnings( "checkstyle:visibilitymodifier" )
+    public String apiguardian;
+
+    @Parameters( name = "{0}" )
+    public static Iterable<Object[]> artifactVersions()
+    {
+        List<Object[]> args = new ArrayList<>();
+        args.add( new Object[] {"1.0.3", "5.0.3", "1.0.0", "1.0.0"} );
+        args.add( new Object[] {"1.1.1", "5.1.1", "1.0.0", "1.0.0"} );
+        args.add( new Object[] {"1.2.0", "5.2.0", "1.1.0", "1.0.0"} );
+        args.add( new Object[] {"1.3.2", "5.3.2", "1.1.1", "1.0.0"} );
+        args.add( new Object[] {"1.4.2", "5.4.2", "1.1.1", "1.0.0"} );
+        args.add( new Object[] {"1.5.2", "5.5.2", "1.2.0", "1.1.0"} );
+        args.add( new Object[] {"1.6.2", "5.6.2", "1.2.0", "1.1.0"} );
+        args.add( new Object[] {"1.7.1", "5.7.1", "1.2.0", "1.1.0" } );
```

Comment:
- only `jupiter` param is used in test ... so we can simplify this arrays
- versions 5.8.x, 5.9.x of jupiter are missing in test

Reply:
ok, I see other tests have the same patterns 😠

Reply:
I mindlessly copied this from JUnitPlatformEnginesIT. I'll simplify it and also reduce the number of versions tested. I guess 5.2.0, 5.8.2 and 5.9.1 should be enough for a bugfix-verifying IT like this. I'll also do that for Surefire1914XmlReportingParameterizedTestIT which is another bugfix-verifying IT I added a while ago.

Reply:
Pushed the changes to Surefire2117XmlReportingNestedIT. 5.8.2 and 5.9.1 are included now. To reduce the runtime of the test I've dropped all other versions except for 5.2.0 which is the earliest JUnit5 version that seems to be widely used.

Did the same changes to Surefire1914XmlReportingParameterizedTestIT, but in another PR (#588) where I also added 5.8.2 and 5.9.1 to JUnitPlatformEnginesIT and JUnitPlatformIT.